### PR TITLE
Convert project to use yarn workspaces

### PR DESCRIPTION
**Pull request by dgeb**
This PR converts this project to use yarn workspaces. This work was motivated by persistent CI failures related to yarn usage with the in-repo addons/engines used to test this project. It's unclear why these failures started, but several other fixes were attempted before resorting to this more extreme step.

### CODE REVIEW

File: packages/ember-engines/.template-lintrc.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+module.exports = {
+  // TODO - "Octanify" templates
```
Comment (rwjblue) [inline packages/ember-engines/.template-lintrc.js:4]: Probably worth making an issue for this

File: packages/ember-engines/node_modules/.bin/_mocha [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/mocha/bin/_mocha
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/_mocha:1]: I don't grok why this is here, can you remove the checked in `packages/ember-engines/node_modules` folder?

File: packages/ember-engines/node_modules/.bin/ember [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-cli/bin/ember
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/ember:1]: Same as above, seems like this file should not be committed (I would expect `yarn install` to regenerated it as needed).

File: packages/ember-engines/node_modules/.bin/ember-source-channel-url [hunk 4]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-source-channel-url/bin/ember-source-channel-url
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/ember-source-channel-url:1]: 🔪

File: packages/ember-engines/node_modules/.bin/ember-template-lint [hunk 5]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-template-lint/bin/ember-template-lint.js
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/ember-template-lint:1]: ✂️

File: packages/ember-engines/node_modules/.bin/eslint [hunk 6]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/eslint/bin/eslint.js
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/eslint:1]: 🍴

File: packages/ember-engines/node_modules/.bin/eslint-config-prettier-check [hunk 7]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/eslint-config-prettier/bin/cli.js
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/eslint-config-prettier-check:1]: 🥒

File: packages/ember-engines/node_modules/.bin/lerna-changelog [hunk 8]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/lerna-changelog/bin/cli.js
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/lerna-changelog:1]: 🔫

File: packages/ember-engines/node_modules/.bin/mocha [hunk 9]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/mocha/bin/mocha
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/mocha:1]: 🧙

File: packages/ember-engines/node_modules/.bin/npm-run-all [hunk 10]
```diff
@@ -0,0 +1 @@
+../../../../node_modules/npm-run-all/bin/npm-run-all/index.js
```
Comment (rwjblue) [inline packages/ember-engines/node_modules/.bin/npm-run-all:1]: 🪓

File: packages/ember-engines/.template-lintrc.js (hunk 1 above)
Comment (dgeb) [reply packages/ember-engines/.template-lintrc.js:4]: Agreed. See #723

**Comment by dgeb** [comment]
@rwjblue thanks for the review! Sorry about including those node_modules ... I've removed them and rebased. ✂️ 

I created #723 to track the template modernization work that's needed.

**Comment by dgeb** [comment]
Looks like there's a failure related to container teardown that's showing in release, beta, and canary. Will have to investigate.

**Pull request merged**